test/tools/ossfuzz/SolidityGenerator.h

```diff
@@ -22,21 +22,210 @@
 
 #pragma once
 
+#include <test/tools/ossfuzz/Generators.h>
+
+#include <liblangutil/Exceptions.h>
+
+#include <memory>
 #include <random>
+#include <set>
+#include <variant>
 
 namespace solidity::test::fuzzer
 {
+/// Forward declarations
+class SolidityGenerator;
+
+/// Type declarations
+#define SEMICOLON() ;
+#define FORWARDDECLAREGENERATORS(G) class G
+GENERATORLIST(FORWARDDECLAREGENERATORS, SEMICOLON(), SEMICOLON())
+#undef FORWARDDECLAREGENERATORS
+#undef SEMICOLON
+
+#define COMMA() ,
+using GeneratorPtr = std::variant<
+#define VARIANTOFSHARED(G) std::shared_ptr<G>
+GENERATORLIST(VARIANTOFSHARED, COMMA(), )
+>;
+#undef VARIANTOFSHARED
+using Generator = std::variant<
+#define VARIANTOFGENERATOR(G) G
+GENERATORLIST(VARIANTOFGENERATOR, COMMA(), )
+>;
+#undef VARIANTOFGENERATOR
+#undef COMMA
```

Comment:
How much code is this sequence of macros replacing? I get what it's doing, but it's not very readable. Is it a lot of code?

Reply:
Essentially, it avoids duplicating the list of generators. I expect this to be about 15-20 generators to cover a fair portion of Solidity grammar. Each list adds about 20 lines of code and I expect about 3-5 lists.

I could do stuff like

```
class Gen1;
class Gen2;
...
class GenN;

...
std::variant<std::shared_ptr<Gen1>,
  std::shared_ptr<Gen2>,
  ...
  std::shared_ptr<GenN>
>
```

instead. Essentially, I am saving about 50-70 lines in the header I guess the more generators I add.

Reply:
Ok!